# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/transmission/hotfix-cmake-4.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- transmission-4.0.6/third-party/dht/CMakeLists.txt.vanilla	2025-07-31 12:19:00.470383926 +0300
+++ transmission-4.0.6/third-party/dht/CMakeLists.txt	2025-07-31 12:19:06.035613164 +0300
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(dht C)
 
 add_library(${PROJECT_NAME} STATIC
--- transmission-4.0.6/third-party/libevent/CMakeLists.txt.vanilla	2025-07-31 12:19:36.140732245 +0300
+++ transmission-4.0.6/third-party/libevent/CMakeLists.txt	2025-07-31 12:19:56.876000000 +0300
@@ -19,7 +19,7 @@
 #       start libevent.sln
 #
 
-cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 
 if (POLICY CMP0054)
     cmake_policy(SET CMP0054 NEW)
--- transmission-4.0.6/third-party/libnatpmp/CMakeLists.txt.vanilla	2025-07-31 12:19:36.143646314 +0300
+++ transmission-4.0.6/third-party/libnatpmp/CMakeLists.txt	2025-07-31 12:20:02.812000000 +0300
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(natpmp C)
 
 add_definitions(-DNATPMP_STATICLIB -DENABLE_STRNATPMPERR)
--- transmission-4.0.6/third-party/libpsl/CMakeLists.txt.vanilla	2025-07-31 12:19:36.146616085 +0300
+++ transmission-4.0.6/third-party/libpsl/CMakeLists.txt	2025-07-31 12:20:07.916000000 +0300
@@ -2,7 +2,7 @@
 # This is just the minimum subset needed for building an embedded
 # static library into Transmission.
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 
 project(psl
   VERSION 0.21.1.0 # when changing this, must set LIBPSL_VERSION_NUMBER too
--- transmission-4.0.6/third-party/rapidjson/CMakeLists.txt.vanilla	2025-07-31 12:19:36.152674408 +0300
+++ transmission-4.0.6/third-party/rapidjson/CMakeLists.txt	2025-07-31 12:20:15.464000000 +0300
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
 if(POLICY CMP0025)
   # detect Apple's Clang
   cmake_policy(SET CMP0025 NEW)
